File: tests/code/corpus/langs/typescript/registry.ts
import { makeCircle } from "./shapes";

export type Slot = { index: number };

export class Registry {
  private items: unknown[] = [];
  add(item: unknown): void {
    this.items.push(item);
  }
  seed(): void {
    this.add(makeCircle(1));
  }
}

export function emptyRegistry(): Registry {
  return new Registry();
}

File: tests/code/corpus/langs/dart/registry.dart
import 'shapes.dart';

class Registry {
  final List<Object> items = [];

  void add(Object item) {
    items.add(item);
  }

  void seed() {
    add(makeCircle(1.0));
  }
}

Registry emptyRegistry() {
  return Registry();
}
